internal/resolver/dns/dns_resolver.go

```diff
@@ -183,6 +191,60 @@ type dnsResolver struct {
 	// has data race with replaceNetFunc (WRITE the lookup function pointers).
 	wg                   sync.WaitGroup
 	disableServiceConfig bool
+
+	polling chan struct{}
+}
+
+// poll begins or ends asynchronous polling of the resolver based on whether
+// err is ErrBadResolverState.
+func (d *dnsResolver) poll(errorFromClientConn error) {
```

Comment:
Something like `updateErr` or `updateStateErr` would be more concise and go-like.

Reply:
Changed to "updateStateErr".